### PR TITLE
fix(webserver): Disallow admins from disabling themselves or changing their own roles

### PR DESCRIPTION
**Pull request by boxbeam**

### CODE REVIEW

File: ee/tabby-webserver/src/schema/mod.rs [hunk 1]
```diff
@@ -394,7 +396,7 @@ impl Mutation {
     async fn update_user_active(ctx: &Context, id: ID, active: bool) -> Result<bool> {
         check_admin(ctx)?;
         if ctx.claims.as_ref().is_some_and(|c| c.sub.0 == id) {
-            return Err(anyhow!("Cannot disable yourself").into());
+            return Err(CoreError::SelfTargetedAction);
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/schema/mod.rs:399]: use `Unauthorized`

File: ee/tabby-webserver/src/schema/mod.rs [hunk 2]
```diff
@@ -403,7 +405,7 @@ impl Mutation {
     async fn update_user_role(ctx: &Context, id: ID, is_admin: bool) -> Result<bool> {
         check_admin(ctx)?;
         if ctx.claims.as_ref().is_some_and(|c| c.sub.0 == id) {
-            return Err(anyhow!("Cannot change your own role").into());
+            return Err(CoreError::SelfTargetedAction);
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/schema/mod.rs:408]: ditto

**Review by wsxiaoys — CHANGES_REQUESTED** [review]

File: ee/tabby-webserver/src/schema/mod.rs (hunk 1 above)
Comment (wsxiaoys) [reply ee/tabby-webserver/src/schema/mod.rs:399]: Sorry, should be `Forbidden`

**Comment by boxbeam** [comment]
Fixes TAB-551

**Pull request merged**